sources/packages/backend/libs/integrations/src/services/sfas/sfas-restriction.service.ts

```diff
@@ -108,6 +124,58 @@ export class SFASRestrictionService
     await this.repo.save(restriction, { reload: false, transaction: false });
   }
 
+  /**
+   * Updates processed to false for the SFAS restrictions corresponding to the provided SFAS Individual Id.
+   * @param individualId provided SFAS Individual Id.
+   */
+  async updateProcessedStatus(individualId: number): Promise<void> {
+    await this.repo.update({ individualId }, { processed: false });
+  }
+
+  /**
+   * Create notifications for newly created legacy student restrictions for the ministry.
+   * @param studentIds student ids for which the legacy restriction notification needs to be sent.
+   * @param entityManager entity manager to execute in transaction.
+   */
+  private async createLegacyRestrictionNotifications(
+    studentIds: number[],
+    entityManager?: EntityManager,
+  ): Promise<void> {
+    const auditUser = this.systemUsersService.systemUser;
+    const studentLegacyRestrictions = await this.studentRestrictionRepo.find({
+      select: {
+        student: {
+          birthDate: true,
+          user: { id: true, firstName: true, lastName: true, email: true },
+        },
+      },
+      relations: { student: { user: true } },
+      where: {
+        restriction: { restrictionCode: "LGCY" },
```

Comment:
I believe that a student ID would be enough to execute the query.

Reply:
The second line of the comment above was suggesting to use the student directly instead of the `StudentRestriction`. The query can be based on `getRepository(StudentRestriction)`, unless I am missing something.